Remotion/Security/Core/Metadata/LocalizationFileNameStrategy.cs

```diff
@@ -27,8 +28,9 @@ public class LocalizationFileNameStrategy
     public string GetLocalizationFileName (string metadataFilename, CultureInfo culture)
     {
       string baseFilename = Path.GetFileNameWithoutExtension (metadataFilename);
-      // TODO RM-7870: Add notnull assertion
-      string basePath = Path.GetDirectoryName (metadataFilename)!;
+      string? basePath = Path.GetDirectoryName (metadataFilename);
+      Assertion.IsNotNull (basePath);
```

Comment:
Please add a message, maybe something of the sort `nameof(metadataFilename) + " must contain a directory name."`? @MichaelKetting

Reply:
See 79d79dad6011aaf8d60c7e29251a69c5e3a363b3. Went with "non-root directory" due to respective docs.

Reply:
Thanks to the acutal code and discussion and need to inject the parameter value, I realized that a) the method is public, b) this is a parameter, and c) the paramter's value is actually invalid if this error occurs. So, we should make this into a straight forward argument exception.